### DISCUSSION TITLE
Easier private sources

### DISCUSSION THREAD

**Discussion by aqua-nerd** [microsoft/winget-cli#2168] [Ideas]
Thought I would have a play with winget this week thinking maybe its come of age. Thought by now private sources would be an easy thing to do. It seems pretty convoluted to get a private source going since its basically just somewhere to host manifest files it should be able to work with any basic web server. I don't see the point in having to spin things up in Azure like Cosmos DB and other things I know nothing about just to host a few packages. Some of us are still on-prem shops and like most other Microsoft projects your leaving these shops behind! Then you have this guy trying to shill his winget.pro service all over the place, its super frustrating. Please make a reference winget rest source that we can run on-prem.

Ok rant over, I'm off to deploy a few packages via GPO like its 2008!

**Comment by denelon** [comment]
We've been discussing a lighter weight approach for on-prem hosting. The [swagger API](https://github.com/microsoft/winget-cli-restsource/tree/main/documentation) defines the REST contracts. I'm believe we could build something using IIS and SQLLite for small instances with a small load of users and packages. The reference implementation over at https://github.com/microsoft/winget-cli-restsource is being designed to support a large number of users and packages.

**Reply by denelon** [reply]
* https://github.com/microsoft/winget-cli-restsource/issues/135

**Comment by mherrmann** [comment]
Hey @aqua-nerd, winget.pro guy here. I don't understand why you're taking offense at it. It's an honest offering. And "shill all over" is honestly an exaggeration. I've probably written 5-10 comments on the entire internet mentioning it. And I believe it's fair in the spirit of the decidedly open ecosystem that Microsoft is creating here. So sorry, but I find your complaint about me to be pretty unwarranted.
